base/src/main/scala/co/uproot/abandon/Config.scala

```diff
@@ -406,7 +407,52 @@ case class BalanceReportSettings(
   showZeroAmountAccounts: Boolean) extends ReportSettings(_title, _accountMatch, _outFiles) {
 }
 
-case class RegisterReportSettings(_title: String, _accountMatch: Option[Seq[String]], _outFiles: Seq[String]) extends ReportSettings(_title, _accountMatch, _outFiles) {
+/**
+  * GroupBy selectors
+  */
+sealed trait GroupBy
+sealed case class GroupByYear() extends GroupBy
+sealed case class GroupByMonth() extends GroupBy
+sealed case class GroupByDay() extends GroupBy
+sealed case class GroupByIsoWeek() extends GroupBy
+sealed case class GroupByIsoWeekDate() extends GroupBy
```

Comment:
Later I like to use these as closed set enums, so that compiler can do exhaustive check in matchers. With objects that's not possible?
Especially here: https://github.com/hrj/abandon/pull/122/files#diff-4943d1460731e0560df4dd1a2f3a6088R444
Throw:n exception while figuring out user input is kind of ok (input/cfg validation), but later inside maps etc. is is not so nice. That was the reason behind sealed case class.